lib/webtask.js

```diff
@@ -218,9 +224,16 @@ exports.issue_token = function (options /* req, claims, kid, keyset, store_code_
             new_claims.ten = new_claims.ten.regex.toString();
     }
 
-    // Encrypt context
 
-    if (new_claims.ectx) {
+    if(new_claims.ectx) {
+        // Token ectx stores authorization values correctly
+
+        if(!(new_claims.ectx.WEBTASK_JWT_SECRET && new_claims.ectx.WEBTASK_JWT_AUD) &&
+            (new_claims.ectx.WEBTASK_JWT_SECRET || new_claims.ectx.WEBTASK_JWT_AUD))
```

Comment:
Also, generally speaking parameter validation of the `issue_token` request is factored out to `validate_token_issue_request`, we should move this check there. In `issue_token` you can assume validation had already happened.